alibi_detect/saving/tensorflow/_loading.py

```diff
@@ -281,9 +241,9 @@ def load_detector_legacy(filepath: Union[str, os.PathLike], suffix: str, **kwarg
         vae = load_tf_vae(filepath, state_dict)
         detector = init_od_vae(state_dict, vae)
     elif detector_name == 'Mahalanobis':
-        detector = init_od_mahalanobis(state_dict)
+        detector = init_od_mahalanobis(state_dict)  # type: ignore[assignment]
```

Comment:
Wondering if we could change the earlier type hint:

```python
 detector = None  # type: Optional[Detector]  # to avoid mypy errors
```

to avoid needing these type ignore's? i.e. change it to `Optional[ALLOWED_DETECTORS]`?

Reply:
`ALLOWED_DETECTORS` (now `VALID_DETECTORS`) is of type: `List[str]` now? Doing the above gives me `mypy` errors. Is that what you meant?

Reply:
Ooops ignore me, I had in my head that `VALID_DETECTORS` was a `Union` of ForwardRef's when I said this. I am happy to leave these type ignore's here, especially since it is a legacy function.